main.py

```diff
@@ -347,3 +347,21 @@ def send_message(message):
         finally:
             # Disconnect from the SMTP server
             smtp.quit()
+
+
+def pubsub(event, context):
+    # Extract the Pub/Sub message data
+    pubsub_message = base64.b64decode(event['data']).decode()
+
+    # Split the message into individual URLs
+    urls = pubsub_message.splitlines()
+
+    # Create an instance of the Cache class
+    cache = Client("cache_file_storage", 5 * 1024 * 1024)
```

Comment:
Please follow the pattern of `get_db_client()`/`get_oo_client()`/etc. to only create the cache client once per instance and only when necessary.

Reply:
Got it